docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,206 @@
+
+The User struct represent both a Go struct as well as the schema for the `users` table on Referral Manager.
+
+```
+type User struct {
+    ID uuid.UUID
+    satelliteURL string
+    NewTokens int
+    UnredeemedTokens int
+}
+```
+
+
+**Endpoints for satellites:**
+
+```
+// GetTokens retrieves a list of unredeemed tokens for a user.
+GetTokens(userID uuid.UUID, satelliteURL string) []tokens {
+    user, err := db.GetUser(userID, satelliteURL)
+    if err == UserNotFound {
+        db.CreateUser(userID, satelliteURL)
+        return nil
+    }
+    if user.NewTokens > 0 {
+        for i:=0; i<user.NewTokens; i++ {
+            newToken := Token{data: generateRandomToken(), user: user.ID, satellite: user.satelliteURL}
+            db.CreateToken(newToken)
+            tokens = append(tokens, newToken)
+        }
+        db.UpdateUserNewTokens(userID, satelliteURL, 0)
+    }
+    tokens := db.GetTokensByUserIDAndSatelliteURL(userID, satelliteURL)
+
+    return tokens
+}
+
+// Redeem marks a token as redeemed and stores user info into database
+func Redeem(ctx, token, userID, satelliteURL) error {
+	// only update the status if the status of a token is unredeems
+	tokenStatus, err := db.RedeemToken(ctx, token, userID)
+	if err != nil {
+		return err
+	}
+
+    // decrease unredeemed token count in users table by 1
+    err := db.UpdateUserInfo(ctx, user)
+    if err != nil {
+        return err
+    }
+
+    // save user info into users table
+    err := db.CreateUser(ctx, userID, satelliteURL)
+    if err != nil {
+        // log the error, but we shouldn't return an error to stop user registration process if we don't get their info here
+        log(err)
+        return
+    }
+}
+```
+
+**Endpoints for CLI:**
+
+```
+
+// GenerateTokens generates tokens, saves those in the referral manager db
+GenerateTokens(tokensPerUser int, satelliteURLs []string, dryRun bool) (tokenCount, eligibleUserCount int, error) {
+    eligibleUsers := db.GetEligibleUsers(satelliteURLs, tokensPerUser)
+    if eligibleUsers == nil {
+        return nil, Error.New("No users to generate tokens for.")
+    }
+    newTokenCount := 0
+    eligibleUserCount := len(eligibleUsers)
+    for _, user := range eligibleUsers {
+        for i:=(user.NewTokens + user.UnredeemedTokens); i<tokensPerUser; i++ {
+            newTokenCount++
+        }
+    }
+
+    if !dryRun {
+        db.UpdateUserNewTokensBatch(eligibleUsers, tokensPerUser)
+    }
+
+    return newTokenCount, eligibleUserCount, nil
+}
+```
+
+**CLI code:**
+```
+startCmd() {
+    tokenCount, userCount := referralManager.GenerateTokens(tokensPerUser, args.SatelliteURLs, flags.dryRun)
+
+    fmt.Printf("Successfully created %d tokens for %d users.\n", tokenCount, userCount)
+    if flags.dryRun {
+        fmt.Println("This was a dry run. Run again without the --dry-run flag to actually generate tokens.")
+    }
+}
+```
+
+## Wrapup
+
+Team Green will be responsible implementing this blueprint.
```

Comment:
Remove the `Team Green` comment since the responsibilities what people do, move around.